plugin/trino-hive/src/test/java/io/trino/plugin/hive/BaseHiveConnectorTest.java

```diff
@@ -4881,6 +4881,35 @@ public void testDropColumn()
         assertUpdate("DROP TABLE test_drop_column");
     }
 
+    @Override
+    public void testDropAndAddColumnWithSameName()
+    {
+        // Override because the behavior depends on orc_use_column_names session property
+        Session session = Session.builder(getSession())
+                .setCatalogSessionProperty(catalog, "orc_use_column_names", "true")
+                .build();
+
+        try (TestTable table = new TestTable(getQueryRunner()::execute, "test_drop_add_column", "AS SELECT 1 x, 2 y, 3 z")) {
+            assertUpdate("ALTER TABLE " + table.getName() + " DROP COLUMN y");
+            assertThatThrownBy(() -> assertQuery("SELECT * FROM " + table.getName(), "VALUES (1, 3)"))
+                    .hasMessageContaining("""
+                            Actual rows (up to 100 of 1 extra rows shown, 1 rows in total):
```

Comment:
Can we pass the values `VALUES (1, 3)` and maybe add a comment on why it is throwing like this ? Shouldn't we avoid these operations or warn the user about the consequences about this ?